### ISSUE TITLE
too much cpu and memory

### ISSUE THREAD

**Issue by jabiers** [zcash/zcash#2641]
I am using aws t2.xlarge ec2 ubuntu 16.04 server.

It is running 3 coin daemons(zcashd, zend, hushd) and z-nomp node service within same server.

zcash has too much usaging resources and it  makes sometme rpc connection  disconnected.

is it normal? or something wrong?
<img width="1214" alt="2017-09-28 8 44 38" src="https://user-images.githubusercontent.com/7077523/30943094-c11421d4-a429-11e7-99fd-5214b25bbcfc.png">

**Comment by ioptio** [comment]
During this time did you create any shielded transactions or were you simply running the daemon keeping it sync? Was the sync catching up at this time?

**Comment by zhangdongbo** [comment]
My zcashd have same issue
After running 10 hours, It occupy 12 GB memory and will halt after running 14 hours with exhausted memory
No transactions and no syncing

```
 PID USER      PR  NI    VIRT    RES    SHR S  %CPU %MEM     TIME+ COMMAND                                                                                                                 
 1074 zec       20   0 13.711g 0.012t  37604 S  35.2 78.2 355:42.98 zcashd 
```

**Comment by ioptio** [comment]
@str4d wondering if the memory benchmarks you added recently could give any insight into this ticket?

**Comment by jabiers** [comment]
@ioptio 
sorry for late reply.
it is used for miningpool.
and it was synchronized all blocks.

**Comment by ioptio** [comment]
@jabiers are you still having memory issues?

**Comment by jabiers** [comment]
@ioptio I increase memory 64gb
zcash memory is around 10gb.

i don't know if it is solved or not. but works find.

**Comment by ioptio** [comment]
This is related to https://github.com/zcash/zcash/issues/1484 & https://github.com/zcash/zcash/issues/2155

**Comment by ioptio** [comment]
Let's keep track of memory consumption tracking in #2155 but we can still keep this open.

**Comment by mms710** [comment]
Closing this because the issue was resolved for the user and we have #2155
